specs/applicationdata/ApplicationData.md

```diff
@@ -0,0 +1,432 @@
+  - [4.3. Packaged app using MachinePath](#43-packaged-app-using-machinepath)
+  - [4.4. Unpackaged app using LocalPath](#44-unpackaged-app-using-localpath)
+- [5. Remarks](#5-remarks)
+  - [5.1. Platform Support](#51-platform-support)
+- [6. API Details](#6-api-details)
+
+This feature provides application data APIs comparable to those in namespaces
+[Windows.Storage](https://learn.microsoft.com/uwp/api/windows.storage) and
+[Windows.Management.Core](https://learn.microsoft.com/uwp/api/windows.management.core) but with
+additional functionality, improved developer experience and performance optimizations.
+
+This feature provides APIs to access and manipulate app data stores.
+
+# 2. Background
+
+Windows provides data stores for packaged applications to store data by user and package family via
+WinRT APIs in the Windows.Storage and Windows.Management.Core namespaces. These API provide the
+means to create, use and manage data in these data stores.
+
+The Application Data APIs were originally introduced in Windows 8. However, as one of the earliest
+WinRT APIs they include some patterns out of step with current API practices and recommendations.
+Windows App SDK offers the opportunity to provide a new generation of APIs in line with the latest
+patterns and recommendations for an improved developer experience as well as functional enhancements
+and improved runtime efficiencies.
+
+Microsoft-internal tasks [22308619](https://task.ms/22308619),
+[30002314](https://task.ms/30002314), [39401257](https://task.ms/39401257),
+[39414843](https://task.ms/39414843), [40050113](https://task.ms/40050113) and
+[48332357](https://task.ms/48332357).
+
+# 3. Description
+
+This API supports app data stores focusing on the following scenarios:
+
+* ApplicationData and ApplicationDataManager functionality
+* *Path properties
+* Reliable access
+* Per-Machine data store
+* Unpackaged app data stores
+
+## 3.1. ApplicationData and ApplicationDataManager functionality
+
+This API provides an equivalent superset of
+[Windows.Storage.ApplicationData](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata),
+with the following notable differences:
+
+* **No roaming functionality** -- We don't provide equivalents to Windows.Storage.ApplicationData's
+  [RoamingSettings](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.roamingsettings),
+  [RoamingStorageQuota](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.roamingstoragequota),
+  [Version](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.version),
+  [SetVersionAsync()](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.setversionasync),
+  [SignalDataChanged()](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.signaldatachanged)
+  and
+  [DataChanged](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.datachanged).
+  See [5.1. Platform Support](#51-platform-support) for more details.
```

Comment:
Also, no SharedLocal functionality (shared per user by all apps from the same publisher). Is that basically the same as the new Machine folder?

Reply:
No. Key difference is Machine = per-machine, not per-user. SharedLocal is ACL'd for access by apps across packages by the publisher, but still managed per-user. Machine isn't per-user.

SharedLocal = per-user per-publisher
Machine = per-machine per-packagefamily

I guess you could say Machine a higher level escalation of Local
* Local = per-user per-packagefamily
* Machine = per-machine per-packagefamily

SharedLocal is a bit different axis from that, as SharedLocal is still per-user but being per-publisher it's cross-packagefamily. Given e.g. FantastyFootball, FantasyHockey and FantasyBaseball package families all by publisher=FabrikamSports &lt;g&gt; and you and I both have all 3 packages registered the machine will have...
* 6 Local paths (3 in your %USERPROFILE%, 3 in mine)
* 3 Machine paths (assuming all 3 packages opt-in via their appxmanifest)
* 2 SharedLocal paths (1 in your %USERPROFILE%, 1 in mine)

So yes, there's some vague similiarities of SharedLocal and Machine in that they offer wider access than just the 1 package family for user, but SharedLocal's still a per-user thing. Machine isn't.

The scenario for Machine is I need to store some files on the machine every user may need to Read but could be sizable, so I don't want every user having their own copy. I can make 1 copy on the machine, ACL'd so only my PackagedService will modify and manage, and everyone else can read and share the 1 copy. In the simple extreme case, a Citrix, VMware or AVD system could have 100 users concurrently logged in using the system. If I use SharedLocal to share these large files I'd have 100 copies. Using Machine I can have 1.